ports/esp32/Makefile

```diff
@@ -256,21 +256,8 @@ $(HEADER_BUILD)/qstrdefs.generated.h: $(SDKCONFIG_H)
 ################################################################################
 # List of object files from the ESP32 IDF components
 
-ESPIDF_DRIVER_O = $(addprefix $(ESPCOMP)/driver/,\
-	uart.o \
-	periph_ctrl.o \
-	ledc.o \
-	gpio.o \
-	timer.o \
-	sdmmc_host.o \
-	sdmmc_transaction.o \
-	sdspi_crc.o \
-	sdspi_host.o \
-	sdspi_transaction.o \
-	spi_master.o \
-	spi_common.o \
-	rtc_module.o \
-	)
+ESPIDF_DRIVER_C = $(wildcard $(ESPCOMP)/driver/*.c)
+ESPIDF_DRIVER_O = $(ESPIDF_DRIVER_C:.c=.o)
```

Comment:
How about combining this into one declaration, to avoid the need to define `ESPIDF_DRIVER_C`?

Reply:
Honestly, I don't know how to get make to do that in one line. It seems to evaluate the replace before the wildcard, and just breaks. 

If you know how, shoot. Otherwise, I'll look in to it further.

Reply:
Try using patsubst, see py/py.mk for examples.

Reply:
Its one line now.

Reply:
`patsubst` needs the whole path to be satisfied which elongates the line by a substantial amount, even with the `$(ESPCOMP)` variable. `subst`, though not as safe, seems perfectly sufficient.

Reply:
Yes it looks ok with `subst`